File: skellig-junit-runner/src/test/java/org/skellig/runner/config/SimpleMessageTestStep.java
package org.skellig.runner.config;

import org.skellig.teststep.processing.model.TestStep;
import org.skellig.teststep.processing.model.TestStepExecutionType;
import org.skellig.teststep.processing.model.ValidationDetails;

import java.util.Map;

public class SimpleMessageTestStep extends TestStep {

    private String receiver;

    protected SimpleMessageTestStep(String id, String name, TestStepExecutionType execution, int timeout, int delay,
                                    Map<String, Object> variables, Object testData, ValidationDetails validationDetails, String receiver) {
        super(id, name, execution, timeout, delay, variables, testData, validationDetails);
        this.receiver = receiver;
    }

    public String getReceiver() {
        return receiver;
    }

    public static class Builder extends TestStep.Builder {

        private String receiver;

        public Builder withReceiver(String receiver) {
            this.receiver = receiver;
            return this;
        }

        @Override
        public TestStep build() {
            return new SimpleMessageTestStep(id, name, execution, timeout, delay, variables, testData, validationDetails, receiver);
        }
    }
}

File: skellig-junit-runner/src/test/kotlin/org/skellig/runner/config/SimpleMessageTestStep.kt
package org.skellig.runner.config

import org.skellig.teststep.processing.model.DefaultTestStep
import org.skellig.teststep.processing.model.ScenarioStateUpdater
import org.skellig.teststep.processing.model.TestStepExecutionType
import org.skellig.teststep.processing.model.ValidationNode

class SimpleMessageTestStep private constructor(id: String?,
                                                name: String?,
                                                execution: TestStepExecutionType,
                                                timeout: Int,
                                                delay: Int,
                                                attempts: Int,
                                                values: Map<String, Any?>?,
                                                testData: Any?,
                                                validationDetails: ValidationNode?,
                                                scenarioStateUpdaters: List<ScenarioStateUpdater>?,
                                                val receiver: String?,
                                                val receiveFrom: String?)
    : DefaultTestStep(id, name!!, execution, timeout, delay, attempts, values, testData, validationDetails, scenarioStateUpdaters) {

    class Builder : DefaultTestStep.Builder<SimpleMessageTestStep>() {

        private var receiver: String? = null
        private var receiveFrom: String? = null

        fun withReceiver(receiver: String?) = apply {
            this.receiver = receiver
        }

        fun withReceiveFrom(receiveFrom: String?) = apply {
            this.receiveFrom = receiveFrom
        }

        override fun build(): SimpleMessageTestStep {
            return SimpleMessageTestStep(id, name, execution, timeout, delay, attempts, values, testData,
                    validationDetails, scenarioStateUpdaters, receiver, receiveFrom)
        }
    }
}